### PR TITLE
Configurable Route Generation

### PR DESCRIPTION
**Pull request by sjaveed**
* friendly_id now takes a routes option which, when set to anything but :friendlyid will generate standard rails style routes when used with e.g. polymorphic_path or form_for etc
* when the routes option isn't specified, it defaults to the current behavior

Given a model:
```ruby
class Team < ActiveRecord::Base
  friendly_id :name, use: :slugged
end
```
it generates:
```ruby
team_path(team)  # returns /team/123-awesome-team-name
team_path(team.id) # returns /team/123
polymorphic_path(team) # returns /team/123-awesome-team-name
```

With this patch, given a model like:
```ruby
class Team < ActiveRecord::Base
  friendly_id :name, use: :slugged, routes: :default
end
```
it generates:
```ruby
team_path(team) # should return /team/123
team_path(team.friendly_id) # should return /team/123-awesome-team-name
polymorphic_path(team) # should return /team/123
```

### CODE REVIEW

**Comment by sjaveed** [comment]
@parndt here you go!

File: lib/friendly_id/base.rb [hunk 1]
```diff
@@ -196,6 +196,7 @@ module Base
     def friendly_id(base = nil, options = {}, &block)
       yield friendly_id_config if block_given?
       friendly_id_config.dependent = options.delete :dependent
+      # friendly_id_config.routes = (options.delete(:routes) || :friendly)
```
Comment (parndt) [inline lib/friendly_id/base.rb]: I'm a little confused about this line of code and why it's commented out? :smile:

**Comment by parndt** [comment]
@sjaveed that was lightning :zap: fast.  

Could we please have some tests to ensure that we don't regress this behaviour, and a note in the changelog file?

File: lib/friendly_id/base.rb (hunk 1 above)
Comment (sjaveed) [inline lib/friendly_id/base.rb]: Good catch :-)  I think I'd added it in there before I realized I could leverage the set.  Commented it out to test the set and then didn't delete it before committing :-)  I'll remove this in the next commit.

**Comment by sjaveed** [comment]
@parndt heh thanks :-)  I did say I was frustrated :-)  Let me see how to test this sucker properly.

**Comment by sjaveed** [comment]
@parndt updated to fix an issue with the default value of routes and to add tests.  Any specific format you'd like me to follow for the Changelog?

**Comment by parndt** [comment]
For changelog just something simple that links to this PR like in https://github.com/norman/friendly_id/blob/master/Changelog.md#520-not-released-yet

**Comment by sjaveed** [comment]
@parndt updated the change log as well.  Do let me know if you need me to change anything else :-)

**Comment by parndt** [comment]
Thanks, I merged it as 8531cdc :smile:

**Comment by sjaveed** [comment]
Awesome!  Thanks @parndt!

**Pull request closed without merging**